src/run/serve.rs

```diff
@@ -1,133 +1,121 @@
-use actix_ws::AggregatedMessage;
-use futures_util::StreamExt as _;
+use std::net::SocketAddr;
+use axum::{
+    extract::{
+        ws::{Message, WebSocket},
+        WebSocketUpgrade,
+    },
+    response::Response,
+    routing::{any, get},
+    Router,
+};
+use tower_http::{
+    services::{ServeDir, ServeFile},
+    trace::TraceLayer,
+};
 
 use crate::web::bundle::{Index, LinkedBundle, PackedBundle, WebBundle};
 
-/// Serve a static HTML file with the given content.
-async fn serve_static_html(content: &'static str) -> impl Responder {
-    // Build the HTTP response with appropriate headers to serve the content as a file
-    HttpResponse::Ok()
-        .insert_header((
-            actix_web::http::header::CONTENT_TYPE,
-            "text/html; charset=utf-8",
-        ))
-        .body(content)
+async fn dev_websocket(ws: WebSocketUpgrade) -> Response {
+    ws.on_upgrade(handle_socket)
 }
 
-/// Serve a static JavaScript file with the given content.
-async fn serve_static_js(content: &'static str) -> impl Responder {
-    // Build the HTTP response with appropriate headers to serve the content as a file
-    HttpResponse::Ok()
-        .insert_header((
-            actix_web::http::header::CONTENT_TYPE,
-            "text/javascript; charset=utf-8",
-        ))
-        .body(content)
-}
-
-/// Handle the websocket connection to the client in dev mode.
-async fn dev_websocket(req: HttpRequest, stream: web::Payload) -> Result<HttpResponse, Error> {
-    let (res, mut session, stream) = actix_ws::handle(&req, stream)?;
-
-    let mut stream = stream
-        .aggregate_continuations()
-        // aggregate continuation frames up to 1MiB
-        .max_continuation_size(2_usize.pow(20));
-
-    rt::spawn(async move {
-        // receive messages from websocket
-        while let Some(msg) = stream.next().await {
-            if let Ok(AggregatedMessage::Text(text)) = msg {
-                // echo text message
-                session.text(text).await.unwrap();
+async fn handle_socket(mut socket: WebSocket) {
+    while let Some(Ok(msg)) = socket.recv().await {
+        if let Message::Text(msg) = msg {
+            if socket
+                .send(Message::Text(format!("You said: {msg}").into()))
+                .await
+                .is_err()
+            {
+                break;
             }
         }
-    });
-
-    // respond immediately with response connected to WS session
-    Ok(res)
+    }
 }
 
 /// Launch a web server running the Bevy app.
-pub(crate) fn serve(web_bundle: WebBundle, port: u16) -> anyhow::Result<()> {
-    rt::System::new().block_on(
-        HttpServer::new(move || {
-            let mut app = App::new();
+#[tokio::main]
+pub(crate) async fn serve(web_bundle: WebBundle, port: u16) -> anyhow::Result<()> {
+    let addr = SocketAddr::from(([127, 0, 0, 1], port));
+    let listener = tokio::net::TcpListener::bind(addr).await.unwrap();
+    tracing::info!("listening on {}", listener.local_addr().unwrap());
```

Comment:
I don't think we need this log, at least not on an `info` level.

We already log "Open your app at <http://localhost:4000>!"